desktop/apps/consignments2/components/sign_up/index.js

```diff
@@ -0,0 +1,118 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import block from 'bem-cn'
+import { Field, reduxForm } from 'redux-form'
+import { compose } from 'underscore'
+import { connect } from 'react-redux'
+import { renderTextInput } from '../text_input'
+import {
+  signUp,
+  updateAuthFormStateAndClearError
+} from '../../client/actions'
+
+function validate (values) {
+  const {
+    email,
+    name,
+    password
+  } = values
+  const errors = {}
+
+  if (!name) errors.name = 'Required'
+  if (!email) errors.email = 'Required'
+  if (!password) errors.password = 'Required'
+
+  return errors
+}
+
+function SignUp (props) {
+  const {
+    error,
+    handleSubmit,
+    signUpAction,
+    invalid,
+    pristine,
+    updateAuthFormStateAndClearErrorAction
+  } = props
+
+  const b = block('consignments2-submission-sign-up')
+
+  return (
+    <div className={b()}>
+      <div className={b('title')}>
+        Create an Account
+      </div>
+      <div className={b('subtitle')}>
+        Already have an account? <span className={b('clickable')} onClick={() => updateAuthFormStateAndClearErrorAction('logIn')}>Log in</span>.
+      </div>
+      <form className={b('form')} onSubmit={handleSubmit(signUpAction)}>
+        <div className={b('row')}>
+          <div className={b('row-item')}>
+            <Field name='name' component={renderTextInput}
+              item={'name'}
+              label={'Full Name'}
+            />
+          </div>
+        </div>
+        <div className={b('row')}>
+          <div className={b('row-item')}>
+            <Field name='email' component={renderTextInput}
+              item={'email'}
+              label={'Email'}
+            />
+          </div>
+        </div>
+        <div className={b('row')}>
+          <div className={b('row-item')}>
+            <Field name='password' component={renderTextInput}
```

Comment:
The new `redux-form` API is way more maintainable; before you had to spread fields in similar to `{...fields.password}` which made making reusable components more verbose than needed.